monai/transforms/smooth_field/array.py

```diff
@@ -217,7 +325,137 @@ def __call__(self, img: np.ndarray, randomize: bool = True):
         field = self.sfield()
         rfield, *_ = convert_to_dst_type(field, img)
 
+        # everything below here is to be computed using the destination type (numpy, tensor, etc.)
+
         out = img * rfield
-        out, *_ = convert_to_dst_type(out, img, img.dtype)
 
         return out
+
+
+class RandSmoothDeform(RandomizableTransform):
+    """
+    Deform an image using a random smooth field and Pytorch's grid_sample.
+
+    The amount of deformation is given by `def_range` in fractions of the size of the image. The size of each dimension
+    of the input image is always defined as 2 regardless of actual image voxel dimensions, that is the coordinates in
+    every dimension range from -1 to 1. A value of 0.1 means pixels/voxels can be moved by up to 5% of the image's size.
+
+    Args:
+        spatial_size: input array size to which deformation grid is interpolated
+        rand_size: size of the randomized field to start from
+        pad: number of pixels/voxels along the edges of the field to pad with 0
+        field_mode: interpolation mode to use when upsampling the deformation field
+        align_corners: if True align the corners when upsampling field
+        prob: probability transform is applied
+        def_range: value of the deformation range in image size fractions, single min/max value  or min/max pair
+        grid_dtype: type for the deformation grid calculated from the field
+        grid_mode: interpolation mode used for sampling input using deformation grid
+        grid_padding_mode: padding mode used for sampling input using deformation grid
+        grid_align_corners: if True align the corners when sampling the deformation grid
+        device: Pytorch device to define field on
+    """
+
+    backend = [TransformBackends.TORCH, TransformBackends.NUMPY]
```

Comment:
As your computation is only on `Tensor`, the `backend` should be: `[TransformBackends.TORCH]`.
All the transforms support `numpy in numpy out` and `tensor in tensor out` so far.

Thanks.

Reply:
Changed.